src/api-v1.js

```diff
@@ -632,6 +632,67 @@ api.declare({
 });
 
 api.declare({
+  method: 'get',
+  route: '/ami-set/:id',
+  name: 'getAmiSet',
+  deferAuth: true,
+  scopes: [
+    ['aws-provisioner:view-ami-set:<amiSetId>'],
+    ['aws-provisioner:manage-ami-set:<amiSetId>'],
+  ],
+  title: 'Get AMI Set',
+  stability:  base.API.stability.stable,
+  description: [
+    'Retreive a copy of the requested AMI set.',
+  ].join('\n'),
+}, async function (req, res) {
+  let id = req.params.id;
+
+  // Authenticate request with parameterized scope
+  if (!req.satisfies({id: id})) {
+    return;
+  }
+
+  let amiSet;
+  try {
+    amiSet = await this.AmiSet.load({id: id});
+  } catch (err) {
+    if (err.code === 'ResourceNotFound') {
+      res.status(404).json({
+        error: err.code,
+        msg: id + ' not found',
+      });
+    } else {
+      throw err;
+    }
+  }
+});
+
+api.declare({
+  method: 'get',
+  route: '/list-ami-sets',
+  name: 'listAmiSets',
+  title: 'List AMI sets',
+  stability:  base.API.stability.stable,
+  description: [
+    'Return a list of AMI sets JSON objects.',
+  ].join('\n'),
+}, async function (req, res) {
+
+  try {
+    let list = await this.AmiSet.listAmiSets();
+    return res.reply(list);
+  } catch (err) {
+    debug('error listing amiSets');
+    debug(err);
+    if (err.stack) {
+      debug(err.stack);
+    }
+    throw err;
+  }
```

Comment:
It looks like elements of `list` here would have a different format than that returned from `getAmiSet`, since `listAmiSets` is using `item.ami.json()`.  The least surprising thing for a user of an API is if the two data structures have the same format.  Using JSON schemas will help enforce this.